src/dotnet/commands/dotnet-compile/NativeCompiler.cs

```diff
@@ -22,16 +23,37 @@ public override bool Compile(ProjectContext context, CompilerCommandApp args)
 
             var managedOutput = outputPaths.CompilationFiles.Assembly;
 
+            // Create the library exporter
+            var exporter = context.CreateExporter(args.ConfigValue);
```

Comment:
So a few questions:
- What happens when the wrong implementations get passed into the compiler? Are you looking specifically for an aot-* RID?
- How are you getting that RID at compile time? Are you defaulting to the current aot-{RID}?